app/controllers/admin/products_controller.rb

```diff
@@ -1,33 +1,31 @@
 class Admin::ProductsController < ApplicationController
   before_action :authenticate_user!
   before_action :set_product, except: [:index, :new, :create]
+  before_action :set_categories, only: [:edit, :new]
 
   def index
     @products= Product.all
   end
 
   def new
     @product = Product.new
-    @categories = ProductCategory.all
   end
 
   def create
-    product_category_id = ProductCategory.find_by(title: params[:product][:product_category_id]).id
-    params[:product][:product_category_id] = product_category_id
     @product = Product.new(product_params)
-    @product.save!
+    if @product.save!
```

Comment:
please check, if record does not save then it will go in else block? as you are saving record with "!".